uwblueprint/extend-a-family
# Login error

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description
* Updated AuthRoutes to first check for valid email and password, and then wrong user type and unverified email
* Added conditional statements in the catch block of AuthRoutes to only print custom error messages if the user is in the Learner login portal
* Updated the getUserByEmail() and signInWithPassword() functions in userService and firebaseRestClient to throw "EMAIL_NOT_FOUND" and "INCORRECT_PASSWORD" custom error messages
* Removed the front-end error message



## Steps to test
In the learner portal, 
1. Sign in with an invalid email (you should get "Email not found. Please try again.")
2. Sign in with a valid email but the wrong password (you should get "Incorrect password. Please try again.")
3. Sign in with a valid email and password but with the wrong role (e.g. in an administrator account - you should get "It looks like you're trying to log in with the wrong account type (${wrongRole}). Please select the correct user type (${correctRole}) and try again.")
3. Sign in with a valid email, password, and the correct role but without being verified (e.g. you should get "Please check your email and click on the link to verify your email.")

In the administrator/facilitator portal, 
1. Sign in with an invalid email (you should get "Invalid login credentials")
2. Sign in with a valid email but the wrong password (you should get "Invalid login credentials")
3. Sign in with a valid email and password but with the wrong role (e.g. in a learner account - you should get "It looks like you're trying to log in with the wrong account type (${wrongRole}). Please select the correct user type (${correctRole}) and try again.")
3. Sign in with a valid email, password, and the correct role but without being verified (e.g. you should get "Please check your email and click on the link to verify your email.")


## What should reviewers focus on?
* The custom error message being printed


## Checklist
